lxd/instance/drivers/driver_qemu.go

```diff
@@ -7450,74 +7460,101 @@ func (d *qemu) getVsockID() (uint32, error) {
 	return uint32(vsockID), nil
 }
 
-// freeVsockID returns true if the given vsockID is not yet acquired.
-func (d *qemu) freeVsockID(vsockID uint32) bool {
-	c, err := lxdvsock.Dial(vsockID, shared.HTTPSDefaultPort)
+// acquireVsockID tries to occupy the given vsock Context ID.
+// If the ID is free it returns the corresponding file handle.
+func (d *qemu) acquireVsockID(vsockID uint32) (*os.File, error) {
+	vsockIDAcquistionLock.Lock()
```

Comment:
Is the locking required now that we are pushing it down into the actual kernel?

Reply:
Mhh good point. I'll test but suspect it's obsolete now. Thanks.

Reply:
You had the right guess, changed as requested.